### ISSUE TITLE
Modifiers with arguments result in Statamic\Modifiers\ModifierNotFoundException

### ISSUE THREAD

**Issue by mstratman** [statamic/cms#7895]
### Bug description

Testing exclusively with `{{ title }}` on both the home page and a collection-show page, it seems that any modifier having arguments throws a `Statamic\Modifiers\ModifierNotFoundException`

Modifiers without arguments work fine.

Switching to 'runtime' version in `config//statamic/antlers.php` seems to work (I haven't yet tested extensively - only discovered this was even a thing by the dropdown in this bug report form.

### How to reproduce

`{{ title | regex_replace(' ', '&nbsp;') }}`

### Environment

```yaml
Environment
Application Name: Statamic
Laravel Version: 9.39.0
PHP Version: 8.0.8
Composer Version: 2.4.4
Environment: local
Debug Mode: ENABLED
URL: diypedals.net.test
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / single
Mail: smtp
Queue: sync
Session: file

Statamic
Addons: 1
Antlers: regex
Stache Watcher: Enabled
Static Caching: Disabled
Version: 3.4.7 Solo

Statamic Addons
rias/statamic-redirect: 2.5.0
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

regex (default)

**Comment by edalzell** [comment]
This syntax is ONLY available on `runtime`: `{{ title | regex_replace(' ', '&nbsp;') }}`.

Are you reporting that that syntax doesn't work w/ `regex`? If so, this is not a bug. The equivalent `regex` syntax would be `{{ title regex_replace="' '|&nbsp;"` or something like that.

**Comment by mstratman** [comment]
Sounds like a documentation issue, then.

What you're describing isn't covered at all on the "[Modifiers](https://statamic.dev/modifiers)" page, the "[Modifier syntax](https://statamic.dev/antlers#modifying-data)" page, or the "[full list of modifiers](https://statamic.dev/reference/modifiers)" reference page.

After reading your reply I visited "[legacy antlers parser](https://statamic.dev/antlers-legacy)" page linked from the ["hot tip" at the top of this page](https://statamic.dev/antlers#modifying-data) and found that syntax. 
But I previously disregarded it because it says 

> For sites running Statamic 3.2 and older you'll need to use the [legacy Antlers parser](https://statamic.dev/antlers-legacy). For all other projects, keep reading. You're in the right place.

I'm running 3.4. I never changed, nor even heard of "regex" vs "runtime" until I tried to file this issue. Perhaps the page should instead say  "if your config/statamic/antlers.php ...." since "sites running Statamic 3.2 and older" is clearly wrong.

**Comment by jasonvarga** [comment]
The line at the top of that page should have probably been closer to "If your site started on 3.3 or earlier", but we'll reword it to point you to your antlers config file.
